### DISCUSSION TITLE
Use iOS shortcut to renew MFA

### DISCUSSION THREAD

**Discussion by jeff47** [steilerDev/icloud-photos-sync#490] [Q&A] [answered]
I am trying to figure out a way to renew the MFA token when it expires.  It seems like an iOS shortcut over SSH using:
`docker exec icloud-photos-sync icloud-photos-sync token`
`docker exec icloud-photos-sync enter_mfa XXXXXX`

However, the `token` command waits for input.  Is there a way to get the token command to exit after it gets a response from the enter_mfa script?

**Comment by steilerDev** [comment, marked as answer]
Hey @jeff47 , you would need to [detach the `token` command](https://www.baeldung.com/linux/detach-process-from-terminal) and then wait for the http server to come online.

Something like this:
```
# Run token command in background and force token refresh
docker exec icloud-photos-sync icloud-photos-sync token --refresh-token &

# Wait for MFA server to come online
until docker exec icloud-photos-sync nc -z localhost 80; do
    echo -n '.'
    sleep 5
done

# Input MFA
docker exec icloud-photos-sync enter_mfa XXXXXX
```

I haven't tested this, but if you've got a working version, feel free to share it so I can add it to the docs :)

**Reply by jeff47** [reply]
Thanks, I didn't realize that eventually the `token` command exits on its own.  I wasn't patient enough.

What works for me on my iPhone is to create a shortcut that runs a script over SSH, that has three parts.

1. Run ssh script to execute `docker exec -d icloud-photos-sync token --refresh-token`
2. Ask user to enter the MFA token
3. Take MFA from step 2 and run an ssh script to execute `docker exec icloud-photos-sync enter_mfa XXXXXX`
